uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/FloorFilterController.qml

```diff
@@ -0,0 +1,686 @@
+            return;
+
+        let selectSite
+        for (var i = 0; i < floorManager.sites.length; ++i) {
+            if (GeometryEngine.intersects(
+                        floorManager.sites[i].geometry.extent,
+                        viewpointCenter.center)) {
+                selectSite = floorManager.sites[i];
+                break;
+            }
+        }
+
+        if (selectSite !== undefined) {
+            setSelectedSiteId(selectSite.siteId);
+        } else if (automaticSelectionMode === FloorFilterController.AutomaticSelectionMode.Always) {
+            setSelectedSiteId("");
+        }
+
+        targetScale = floorManager.facilityLayer ? floorManager.facilityLayer.minScale : 0
+        if (targetScale === 0)
+            targetScale = 1500;
+
+        if(viewpointCenter.targetScale > targetScale)
+            return;
+
+        let selectFacility
+        for (i = 0; i < floorManager.facilities.length; ++i) {
+            let facility;
+            if (GeometryEngine.intersects(
+                        floorManager.facilities[i].geometry.extent,
+                        viewpointCenter.center)) {
+                selectFacility = floorManager.facilities[i];
+                break;
+            }
+        }
+        if (selectFacility !== undefined) {
+            setSelectedFacilityId(selectFacility.facilityId);
+        } else if (automaticSelectionMode === FloorFilterController.AutomaticSelectionMode.Always) {
+            setSelectedFacilityId("");
+            setSelectedLevelId("");
+        }
+    }
+
+    /*!
+      \qmlproperty GeoModel geoModel
+      \brief The geomodel the controller is listening for \c FloorManager changes.
+    */
+    property GeoModel geoModel: {
+        // check geoView has been set
+        if(!geoView)
+            return null;
+        if (geoView instanceof MapView)
+            return geoView.map;
+        else if (geoView instanceof SceneView)
+            return geoView.scene;
+        return null;
+    }
+
+    onGeoModelChanged: {
+        // make sure that the geomodel is loaded.
+        geoModel.load();
+    }
+
+    /*!
+      \qmlproperty int updateLevelsMode
+      \brief The mode that defines how levels are made visible/invisible in the geoView.
+      \value UpdateLevelsMode.AllLevelsMatchingVerticalOrder When a level is selected, all levels with matching vertical order are made visible, all other levels are invisible.
+
+      Default is \c{UpdateLevelsMode.AllLevelsMatchingVerticalOrder}
+    */
+    property int updateLevelsMode: FloorFilterController.UpdateLevelsMode.AllLevelsMatchingVerticalOrder
+
+    /*!
+      \qmlproperty FloorManager floorManager
+      \brief The FloorManager associated to the \l geoModel.
+      It can be null if no associated geoModel is available or is not loaded.
+    */
+    property FloorManager floorManager : geoModel? geoModel.floorManager : null
+
+    onFloorManagerChanged: {
+        internal.onFloorManagerLoaded();
```

Comment:
No, `Changed` means that a new instance of `floorManager` has been assigned.
The `onFloorManagerLoaded` is a function created (not an handler) that load the floorManager if not loaded yet, retry load and starts the `FloorFilter` if it is all loaded correctly.
Might change the `onLoaded` name because it is not always loaded, like `floorManagerLoadHandler()`?